### DISCUSSION TITLE
Automatic Amazon captcha recognition?

### DISCUSSION THREAD

**Discussion by savandriy** [jef/streetmerchant#2426] [Ideas and feature requests]
Hey!

I've started using the streetmerchant recently, and noticed that the Amazon captchas are solved manually.

There are open source libraries that solve Amazon captchas:

- https://github.com/dominhhai/captcha-breaker (machine learning)
- https://github.com/a-maliarov/amazoncaptcha (image recognition)

I'm sure there are a few more.

I do have knowledge of Python & JS and will probably be able to port over the code.

What do you think?
Would a PR be accepted for such feature?

**Comment by ArmaanCPro** [comment]
Hi! Have you used these CAPTCHA solvers? If so, does it work? If it does I'm sure @jef would try to add them at some point.

**Reply by savandriy** [reply]
I have tried the https://github.com/a-maliarov/amazoncaptcha one, and it works. But it's in Python, and streetmerchant is in JavaScript. I've started creating a JS version of the library and got ~50-75% of the work done, but am loaded at work atm, so will get back to it when I get a chance. Really want a >3060 gpu, stuck with a 1060@1440p 😢

**Reply by ArmaanCPro** [reply]
Yeah I really want a >1060 gpu lol. Great work and thank you for converting the Python library to JS.
